app/controllers/import_controller.rb

```diff
@@ -26,5 +34,39 @@ def import
       end
     end
   end
-end
-  
+
+  def import_portal_activity
+    request_json = JSON.parse "#{request.body.read}", :symbolize_names => true
+
+    unless User.find_by_email request_json[:activity][:user_email]
+      user = User.create(
+        email: request_json[:activity][:user_email],
+        password: User.get_random_password,
+        is_author: true
+      )
+    end
+
+    portal = Concord::AuthPortal.portal_for_url(request_json[:portal_url]) 
+    authentication = Authentication.find_by_provider_and_uid(portal.strategy_name,request_json[:domain_uid])
+    user = User.find(authentication.user_id)
```

Comment:
Which user is this? Is the admin that is triggering this import? Or is the author of the activity? If it is the author then this won't work because there won't be an Authentication yet.  If it is the admin, then this will mean the external activity in the portal will be owned by the admin instead of the author.  Perhaps you addressed these issues on the portal side of things.  I'm going to merge this in, but please add a comment here clarifying which user this is since I there are two possibilities.